### PR TITLE
started working on associate-repo feature

### LINKED ISSUE #15: Associate repository to a channel
We need a command that associates a Github repository to a channel (maybe don't bother about private repos for now)

### PR DESCRIPTION
**Pull request by diananova**
- not finished yet
- working on local JSON file for now

### CODE REVIEW

**Review by Laurell876 — APPROVED** [review]
Seems good

**Comment by gmelodie** [comment]
@diananova quick reminder if your PR is not finished you can make it a Draft PR

File: bot.py [hunk 1]
```diff
@@ -15,14 +16,33 @@
...
+with open('repos.json') as f:
+  reposDict = json.load(f)
+
+  
+
+@bot.command(brief="associate a repository to this channel")
+async def associate(ctx, repoName): #!git associate repo
+	guild = ctx.message.guild.name
+	channel = ctx.message.channel.id
+	reposDict[channel]=repoName
```
Comment (gmelodie) [inline bot.py]: I wonder how dangerous this dict being global is, especially if we're dealing with `async` functions. Perhaps making a function `update_reposDict` that [locks](https://stackoverflow.com/a/25799871/6032037) the variable could be a better way to go

File: bot.py [hunk 2]
```diff
@@ -15,14 +16,33 @@
...
-
+#TODO: make repoName argument optional 
 @bot.command(brief="brief summary of repo: number of issues, prs, stars, etc. Args: repoName")
 async def summary(ctx, repoName):  #!git summary MLH-Fellowship/github-discord-bot
-	repo = git.get_repo(repoName)
+	channel = str(ctx.message.channel.id)
+	if channel in reposDict:
+		repo=git.get_repo(reposDict[channel])
+	else:
+		repo = git.get_repo(repoName)
```
Comment (gmelodie) [inline bot.py]:
```suggestion
	if channel not in reposDict:
		repoName = reposDict[channel]
	repo = git.get_repo(repoName)
```

**Review by gmelodie — CHANGES_REQUESTED** [review]
Playing with `async` stuff can get dangerous sometimes :smile:

**Comment by diananova** [comment]
- seems to work in deployed version as well.

**Pull request merged**